core/templates/components/entity-creation-services/skill-creation.service.ts

```diff
@@ -16,49 +16,56 @@
-  '$rootScope', '$timeout', '$window', 'AlertsService',
-  'SkillCreationBackendApiService', 'UrlInterpolationService',
-  'EVENT_TOPICS_AND_SKILLS_DASHBOARD_REINITIALIZED',
-  function(
-      $rootScope, $timeout, $window, AlertsService,
-      SkillCreationBackendApiService, UrlInterpolationService,
-      EVENT_TOPICS_AND_SKILLS_DASHBOARD_REINITIALIZED) {
-    var CREATE_NEW_SKILL_URL_TEMPLATE = (
-      '/skill_editor/<skill_id>');
-    var skillCreationInProgress = false;
+import { UrlInterpolationService } from
+  'domain/utilities/url-interpolation.service.ts';
+import { AlertsService } from 'services/alerts.service';
+import { SkillCreationBackendApiService } from
+  'domain/skill/skill-creation-backend-api.service.ts';
+import { LoaderService } from 'services/loader.service';
 
-    return {
-      createNewSkill: function(
-          description, rubrics, explanation, linkedTopicIds) {
-        if (skillCreationInProgress) {
-          return;
-        }
-        for (var idx in rubrics) {
-          rubrics[idx] = rubrics[idx].toBackendDict();
-        }
-        skillCreationInProgress = true;
-        AlertsService.clearWarnings();
-        SkillCreationBackendApiService.createSkill(
-          description, rubrics, explanation, linkedTopicIds)
-          .then(function(response) {
-            $timeout(function() {
-              $rootScope.$broadcast(
-                EVENT_TOPICS_AND_SKILLS_DASHBOARD_REINITIALIZED, true);
-              skillCreationInProgress = false;
-              $window.open(
-                UrlInterpolationService.interpolateUrl(
-                  CREATE_NEW_SKILL_URL_TEMPLATE, {
-                    skill_id: response.skillId
-                  }), '_blank'
-              );
-            }, 150);
-          });
-      }
-    };
+interface SkillCreationResponse {
+  skillId: string;
+}
+
+@Injectable({
+  providedIn: 'root'
+})
+export class SkillCreationService {
+  CREATE_NEW_SKILL_URL_TEMPLATE = '/skill_editor/<skill_id>';
+  skillCreationInProgress = false;
+
+  constructor(
+    private alertsService: AlertsService,
+    private loaderService: LoaderService,
+    private skillCreationBackendApiService: SkillCreationBackendApiService,
+    private urlInterpolationService: UrlInterpolationService,
+  ) { }
+
+  createNewSkill(description, rubrics, explanation, linkedTopicIds) {
+    if (this.skillCreationInProgress) {
+      return;
+    }
+    for (let idx in rubrics) {
+      rubrics[idx] = rubrics[idx].toBackendDict();
+    }
+    this.skillCreationInProgress = true;
+    this.alertsService.clearWarnings();
+    this.loaderService.setLoadingMessage('Creating skill');
+    this.skillCreationBackendApiService.createSkill(
+      description, rubrics, explanation, linkedTopicIds
+    ).then((response: SkillCreationResponse) => {
+      // TODO(srijanreddy98): Replace window with angular router
+      window.location.href = this.urlInterpolationService.interpolateUrl(
```

Comment:
Note that the previous code opens in a new tab.

Reply:
I checked it didn't in my case. In order to open in a new tab `window.open` is used. I cross-checked with develop branch of my fork, for me it didn't open a new tab. However is opening in a new tab is desired functionality, I will change it to `window.open`.

Reply:
In the original code, _blank should open a new window. See https://developer.mozilla.org/en-US/docs/Web/API/Window/open